src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/DebugDirectoryNode.cs

```diff
@@ -88,8 +89,30 @@ int GetNumDebugDirectoryEntriesInModule()
             if (_module == null)
                 return 0;
 
-            ImmutableArray<DebugDirectoryEntry> entries = _module.PEReader.ReadDebugDirectory();
-            return entries == null ? 0 : entries.Length;
+            PEReader moduleReader = _module.PEReader;
+            ImmutableArray<DebugDirectoryEntry> entries = ImmutableArray<DebugDirectoryEntry>.Empty;
+
+            try
+            {
+                entries = moduleReader.ReadDebugDirectory();
+            }
+            catch (BadImageFormatException e)
+            {
+                if (e.Message.Equals("Invalid directory size."))
```

Comment:
not sure if this should be special cased, since on non-english locales this would probably fail anyways

Reply:
I special cased this as well because it failed otherwise with the same BadImageException.

Reply:
Since `e.Message` will be localized, this check will not work for any language except English.